engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/container/MultiInstanceBodyProcessor.java

```diff
@@ -174,14 +181,16 @@ public void afterExecutionPathCompleted(
 
     final var loopCharacteristics = element.getLoopCharacteristics();
 
-    final Either<Failure, Boolean> failureBooleanEither =
+    final Either<Failure, Boolean> canBeCompletedOrFailure =
         canBeCompletedWithCompletionCondition(element, flowScopeContext);
-    if (failureBooleanEither.isLeft()) {
-      incidentBehavior.createIncident(failureBooleanEither.getLeft(), flowScopeContext);
+    if (canBeCompletedOrFailure.isLeft()) {
+      incidentBehavior.createIncident(canBeCompletedOrFailure.getLeft(), flowScopeContext);
       return;
-    } else if (failureBooleanEither.isRight() && failureBooleanEither.get()) {
+    } else if (canBeCompletedOrFailure.isRight() && canBeCompletedOrFailure.get()) {
+      // terminate all remaining child instances because the completion condition evaluates to true.
       stateTransitionBehavior.terminateChildInstances(flowScopeContext);
       if (loopCharacteristics.isSequential()) {
+        // complete the multi-instance body immediately
         stateTransitionBehavior.completeElement(flowScopeContext);
       }
       return;
```

Comment:
❌  I think we've forgotten 1 edge case. Consider a parallel multi-instance where only on the last completed child instance, the completion condition is met.  In that case, there are no children to terminate and the loopCharacteristics is not sequential, so we don't shortcut the completion of the multi-instance and just `return` here. So we never complete the multi-instance body using line 222.